dockerfile

```diff
@@ -0,0 +1,74 @@
+FROM ubuntu:24.04 as apt-requirements
+
+# Install APT requirements.
+RUN --mount=target=/requirements,type=bind,source=requirements <<EOF
+
+    set -ex
+
+    apt update
+
+    apt --yes --no-install-recommends install $(grep -vE "^\s*#" /requirements/requirements.system.txt  | tr "\n" " ")
+
+    apt clean && rm -rf /var/lib/apt/lists/*
+
+EOF
+
+# Compile Google Test.
+FROM apt-requirements as compile-google-test
+
+ARG GOOGLETEST_VERSION
+
+RUN <<EOF
+
+    set -ex
+
+    git clone --depth=1 --branch=v${GOOGLETEST_VERSION} https://github.com/google/googletest googletest
+
+    cd googletest
+
+    cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/opt/google-test-${GOOGLETEST_VERSION}
+
+    cmake --build build
+
+    cmake --build build --target install
+
+EOF
+
+# Compile Kokkos.
+FROM apt-requirements as compile-kokkos
+
+ARG KOKKOS_VERSION
+ARG KOKKOS_PRESET
+
+COPY kokkos.cmake.presets.json /kokkos.cmake.presets.json
+
+RUN <<EOF
+
+    set -ex
+
+    git clone --depth=1 --branch=${KOKKOS_VERSION} https://github.com/kokkos/kokkos kokkos
+
+    cd kokkos
+
+    cp /kokkos.cmake.presets.json CMakePresets.json
+
+    cmake -S . --preset=${KOKKOS_PRESET} -DCMAKE_INSTALL_PREFIX=/opt/kokkos-${KOKKOS_VERSION}/${KOKKOS_PRESET}
+
+    cmake --build --preset=${KOKKOS_PRESET}
+
+    cmake --build --preset=${KOKKOS_PRESET} --target=install
+
+EOF
+
+# Build the final image from previous stages.
+FROM apt-requirements as final
+
+ARG GOOGLETEST_VERSION
+ARG KOKKOS_VERSION
+ARG KOKKOS_PRESET
+
+COPY --from=compile-google-test /opt/google-test-${GOOGLETEST_VERSION}         /opt/google-test-${GOOGLETEST_VERSION}
+COPY --from=compile-kokkos      /opt/kokkos-${KOKKOS_VERSION}/${KOKKOS_PRESET} /opt/kokkos-${KOKKOS_VERSION}/${KOKKOS_PRESET}
+
+ENV GTest_ROOT=/opt/google-test-${GOOGLETEST_VERSION}
+ENV Kokkos_ROOT=/opt/kokkos-${KOKKOS_VERSION}/${KOKKOS_PRESET}
```

Comment:
Reading

- https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry
- https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images

we could also add a `org.opencontainers.image.source` label:

```
LABEL org.opencontainers.image.source=https://github.com/uliegecsm/kokkos-utils
```

(and perhaps other labels too)

Reply:
There is also a label `org.opencontainers.image.licenses`. So we should probably think about the license again :) First option is probably to use the same one as Kokkos?

Reply:
I've used the Apache v2 license, this is the closest I could find.

And done for adding the label to the image.